### PR TITLE
Giveaway Responses & Media Share

### PR DESCRIPTION
**Pull request by AnkhHeart**
- Implemented Media Request Settings
- Implemented Media Request Default Commands
- Fixed Text Area issue that allowed linebreaks with blockReturns set to true
- Removed uencessary files
- Added missing giveaway responses
- Improved command searching using computed prop
- Loyalty data clearing
- User Removal for Loyalty

### CODE REVIEW

File: app/components/page-components/Chatbot/ChatbotDefaultCommands.vue.ts [hunk 1]
```diff
@@ -3,19 +3,21 @@ import { Component } from 'vue-property-decorator';
 import { IDefaultCommand } from 'services/chatbot';
 import { $t } from 'services/i18n';
 import CollapsibleSection from 'components/shared/CollapsibleSection.vue';
+import { mapValues, pickBy, keys } from 'lodash';
```
Comment (gettinToasty) [inline app/components/page-components/Chatbot/ChatbotDefaultCommands.vue.ts]: can you import these separately so we can properly tree-shake the lodash library? also `keys` is available to us as simply `Object.keys`, no need for a helper here

File: app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  .window-toggle__icon {
+    .margin-left();
+  }
+}
+
+.night-theme {
+  .window-toggle__wrapper {
+    background-color: @night-primary;
+    border-color: @night-border;
```
Comment (gettinToasty) [inline app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue]: please use the new scoped variables for all of your color usage in these components, check out `_colors.less` for more info

File: app/components/page-components/Chatbot/ChatbotLoyalty.vue [hunk 3]
```diff
@@ -79,6 +86,22 @@
...
+        @cancel="onCancelHandler"
+        :title="$t('Reset Loyalty Database')"
+        :message="$t('Are you sure you want to reset your entire Loyalty database? This action is irreversible.')"
+      />
+      <ChatbotGenericModalWindow
+        :name="DELETE_LOYALTY_MODAL"
+        :warn="true"
+        @ok="onDeleteHandler"
+        @cancel="onCancelHandler"
+        :title="loyaltyToDelete ?  $t(`Delete ${loyaltyToDelete.viewer.name}'s Loyalty`) : ''"
```
[the remark this answers is not in the document]
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/ChatbotLoyalty.vue]: Done

File: app/components/page-components/Chatbot/ChatbotLoyalty.vue [hunk 4]
```diff
@@ -79,6 +86,22 @@
...
+        :message="loyaltyToDelete ?$t(`Are you sure you want to delete ${loyaltyToDelete.viewer.name}'s loyalty? This action is irreversible`) : ''"
```
[the remark this answers is not in the document]
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/ChatbotLoyalty.vue]: Done

File: app/components/page-components/Chatbot/ChatbotLoyalty.vue.ts [hunk 5]
```diff
@@ -66,6 +83,14 @@ export default class ChatbotLoyalty extends ChatbotBase {
     this.chatbotApiService.Loyalty.addToAll(value);
   }
 
+  onResetHandler() {
+    this.chatbotApiService.Loyalty.clear();
+  }
+
+  onDeleteHandler() {
+    this.chatbotApiService.Loyalty.delete(this.loyaltyToDelete.id);
```
[the remark this answers is not in the document]
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/ChatbotLoyalty.vue.ts]: Done

File: app/components/page-components/Chatbot/windows/ChatbotGenericModalWindow.vue [hunk 6]
```diff
@@ -1,58 +1,55 @@
...
-          <VFormGroup
-            :title="$t(message)"
-            v-model="value"
-            :metadata="valueMetaData"
-          />
-        </div>
-        <div v-else>
-          <h2 v-if="header">{{ $t(header) }}</h2>
-          <p v-if="message">{{ $t(message) }}</p>
+  <modal :name="name" :height="'auto'" :maxHeight="600">
```
[the remark this answers is not in the document]
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/windows/ChatbotGenericModalWindow.vue]: Done

File: app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue.ts [hunk 7]
```diff
@@ -17,70 +17,61 @@ import { debounce } from 'lodash-decorators';
...
-  securityDescription = $t(
-    `This slider helps you filter shared media before it can be submitted.\n' +
-      '1: No security\n
-      '2: 65%+ rating, 5k+ views\n
-      '3: 75%+ rating, 40k+ views\n
-      '4: 80%+ rating, 300k+ views\n
-      '5: 85%+ rating, 900k+ views`,
-  );
-
+  securityDescription = $t(`This slider helps you filter shared media before it can be submitted.\r\n
```
[the remark this answers is not in the document]
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue.ts]: Done

File: app/components/page-components/Chatbot/ChatbotDefaultCommands.vue.ts (hunk 1 above)
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/ChatbotDefaultCommands.vue.ts]: import separately as in import {mapvalues} from 'lodash' and {pickBy} from 'lodash' both on separate liens?

File: app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue (hunk 2 above)
Comment (AnkhHeart) [reply app/components/page-components/Chatbot/windows/ChatbotMediaRequestPreferencesWindow.vue]: Done

**Pull request merged**